packages/runtime/container-runtime/src/summary/summarizerNode/summarizerNode.ts

```diff
@@ -106,12 +111,41 @@ export class SummarizerNode implements IRootSummarizerNode {
 			0x1a0 /* "Already tracking a summary" */,
 		);
 
+		let nodes = 1;
+		let invalidNodes = 0;
+		const sequenceNumberMismatchKeySet = new Set<string>();
+
+		const nodeLatestSummaryRefSeqNum = this._latestSummary?.referenceSequenceNumber;
+		if (
+			nodeLatestSummaryRefSeqNum !== undefined &&
+			latestSummaryRefSeqNum !== nodeLatestSummaryRefSeqNum
+		) {
+			invalidNodes++;
+			sequenceNumberMismatchKeySet.add(
+				`${latestSummaryRefSeqNum}-${nodeLatestSummaryRefSeqNum}`,
+			);
+		}
+
 		this.wipSummaryLogger = summaryLogger;
 
 		for (const child of this.children.values()) {
-			child.startSummary(referenceSequenceNumber, this.wipSummaryLogger);
+			const childStartSummaryResult = child.startSummary(
+				referenceSequenceNumber,
+				this.wipSummaryLogger,
+				latestSummaryRefSeqNum,
+			);
+			nodes += childStartSummaryResult.nodes;
+			invalidNodes += childStartSummaryResult.invalidNodes;
+			for (const invalidSequenceNumber of childStartSummaryResult.mismatchNumbers) {
```

Comment:
Should be more performant to pass the set down via startSummary for children to add to, rather than re-enumerating the children's results over and over again? Could potentially make a difference for a large summary?  e.g. worst case of full-tree summary for a doc with 10000+ of nodes.  Maybe would only ammount to the order of seconds, not sure.  I guess the tree depth is only 3, so wouldn't repeat more than that...

Up to you, thinking out loud.

Reply:
Maybe just pass the whole `IStartSummaryResult` object around and have the children update all 3 parts, for consistency?  Btw this would work only because the tree traversal is done serially (no parallelism).

Reply:
I agree, the only issue is that I would need to further change the `startSummary` api, and adding something like `IStartSummaryResult` may be more confusing than just `latestSummaryRefSeqNum`